### PR TITLE
add the in-memory memorystream and the loaders package

### PR DESCRIPTION
**Pull request by tobowers**
Adds an in-memory scored MemoryStream to the repo which is useful for lots of usecases around keeping track of dialog, choosing appropriate conversational programs, etc.

### CODE REVIEW

File: memory/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "socialagi-memory",
```
Comment (kafischer) [inline memory/package.json:2]: @socialagi/memory ?

**Review by kafischer — CHANGES_REQUESTED** [review]

File: memory/package.json (hunk 1 above)
Comment (tobowers) [reply memory/package.json]: yup!

**Comment by tobowers** [comment]
@kafischer I added the loaders package to this PR too.

**Comment by tobowers** [comment]
changed loaders -> tools

**Review by kafischer — APPROVED** [review]

**Pull request merged**